### ISSUE TITLE
Issue with SMARTS conversion and similarity matching

### ISSUE THREAD

**Issue by LivC193** [rdkit/rdkit#3120]
RDKit version :'2019.03.2'
Operating system: Ubuntu 18.04.3 LTS
-Python version: 3.7
- Are you using conda?: No
Hi sorry to bother you, I don't know if this is an actual issue or if I am doing something wrong but:
```
from rdkit import Chem
molecule1smiles = 'NC1=NC(=O)N(C2CC(O)C(CO)O2)C(O)C1O'
molecule2smiles = 'C1=CN(C(=O)N=C1N)C2C(C(C(O2)CO)O)O'
mol1 = Chem.MolFromSmiles(molecule1smiles)
mol2 = Chem.MolFromSmiles(molecule2smiles)
#Now I want to convert them into SMARTS
molecule1smarts = Chem.MolToSmarts(mol1)
molecule2smarts = Chem.MolToSmarts(mol2)
print('SMARTS molecule 1:', molecule1smarts)
print('SMARTS molecule 2:', molecule2smarts)
```
The output is:
```
SMARTS molecule 1: [#7]-[#6]1=[#7]-[#6](=[#8])-[#7](-[#6]2-[#6]-[#6](-[#8])-[#6](-[#6]-[#8])-[#8]-2)-[#6](-[#8])-[#6]-1-[#8]
SMARTS molecule 2: [#6]1:[#6]:[#7](:[#6](=[#8]):[#7]:[#6]:1-[#7])-[#6]1-[#6](-[#6](-[#6](-[#8]-1)-[#6]-[#8])-[#8])-[#8]
```
The problem is that the aromatic ring bonds are represented as single bonds( '-') instead of aromatic bonds (':') for the first molecule, which is a problem because I want to do pattern matching. This issue persists (it is even worse actually) even if I consider the initial SMILES strings  as SMARTS and do ```Chem.MolFromSmarts(molecule1smiles)```` the output is:
```
'NC1=NC(=O)N(-,:C2CC(-,:O)C(-,:CO)O2)C(-,:O)C1O'
'C1=CN(-,:C(=O)N=C1N)C1C(-,:C(-,:C(-,:O1)CO)O)O'
```
The main issue is that if you do similarity between those 2 structures you get a pretty low score although they are very closely related:
```
molecule1smarts =  Chem.MolFromSmarts(molecule1smarts)
molecule2smarts =  Chem.MolFromSmarts(molecule2smarts)
DataStructs.cDataStructs.SokalSimilarity(Chem.RDKFingerprint(molecule1smarts),Chem.RDKFingerprint(molecule2smarts))
```
Also I am trying to identify if a smiles contains a ring is there any function that can do that in rdkit ? 
I found this function: FraggleSim.select_fragments(fragments, ftype, hac)
but I have no idea what type of data fragments is, ftype shold be FraggleSim.FTYPE_CYCLIC, but I have no idea what hac is. This line does not return anything:
```
FraggleSim.select_fragments('*c1ccccc1*.*cccc*', FraggleSim.FTYPE_CYCLIC, 10)
```

**Comment by greglandrum** [comment]
@LivC182 There are a few things going on here.

1) The first molecule is not aromatic. This is why you aren't seeing aromatic bonds in the SMARTS.
Here's the molecule:
![image](https://user-images.githubusercontent.com/540511/80697211-f10dbd80-8ad8-11ea-8c9e-7396b52af5ba.png)
that doesn't contain an aromatic ring.

2) What you're seeing in the SMARTS is correct: an unspecified bond in SMARTS is "single or aromatic", so you see plenty of single-or-aromatic bond queries - `-,:` - in your SMARTS.

3) generating fingerprints for molecules that have been built from SMARTS very rarely makes sense. The RDKit's fingerprinting algorithms don't in general deal particularly well with the query features that show up in molecules constructed from SMARTS.
Why do you want to calculate the similarity between these queries?

**Comment by LivC182** [comment]
Yes, sorry for the trouble I later realised my 1st molecule was not aromatic. I am trying to do write some functions to identify analogs of nucleotides so a substructure match is not comprehensive enough. The only way I could think so is a similarity search between nucleotides and candidate smiles. I compared the similarity results for all possible combinations (smiles-smiles, smiles-smarts, smarts-smarts) and the best one was smarts-smarts. I also tried all the possible similarity searches (dice, cosine, rogot etc) and the best one in discriminating between true positives and false positives was sokal. Any thoughts ?
